cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1/types.go

```diff
@@ -70,6 +72,18 @@ type AppRepositoryCustomCA struct {
 	SecretKeyRef corev1.SecretKeySelector `json:"secretKeyRef,omitempty"`
 }
 
+// FilterRulesSpec defines a set of rules and aggreagation logic
+type FilterRulesSpec struct {
+	AnyOf []FilterRule `json:"anyOf,omitempty"`
+	AllOf []FilterRule `json:"allOf,omitempty"`
+}
+
+// FilterRule defines a rule
+type FilterRule struct {
+	JQ        string            `json:"jq"`
+	Variables map[string]string `json:"variables,omitempty"`
+}
```

Comment:
The struct here is unexpected for me. I assumed we'd just have a single jq query and currently two possible lists of names... something like:
```go
type FilterRulesSpec struct {
    JQ           string                   `json:"jq"`
    Variables map[string]string `json:"variables,omitempty"`
}
```
where for OCI charts, `Variables` would always include a `package_names` var. 

Or why do you think separate `JQ` query would be needed for the all of and anyof cases, given that we can do a simple ` or ` or ` and ` with jq. Something like `select(.name|test("(pkg1,pkg2,pkg3)") or .name|test("!(pkg4,pkg5,pkg6)")`. (or something, I'd need to check).

Normally I assume the Kubeapps ui would not even required this - it would just be setting a very simple query - "include if the name matches any of the following" or "exclude if the name matches any of the following", combining the entered names into the `(pkg1,pkg2,pkg3)` format. But I just mean that even for the most complicated jq queries we might need in the future (or users might set manually on the AppRepo resource), I think just one JQ query should suffice?

Reply:
yep, with the JQuery format it's true that the `allOf` or `anyOf` formats are less relevant I just added them because it was in the design document but again it can be simplified

Reply:
Great, thanks.